server/src/main/java/org/elasticsearch/repositories/FinalizeSnapshotContext.java

```diff
@@ -120,4 +135,22 @@ public void onDone() {
     public void onResponse(RepositoryData repositoryData) {
         delegate.onResponse(repositoryData);
     }
+
+    /**
+     * A record used to track the new shard generations that have been written for each shard in a snapshot.
+     * An index may be deleted after the shard generation is written but before the snapshot is finalized.
+     * In this case, its shard generation is tracked in {@link #deletedIndices}. Otherwise, it is tracked in
```

Comment:
I think a little more detail might help in future here, something like this?

```suggestion
     * In this case, its shard generation is tracked in {@link #deletedIndices} because it's still a valid shard generation blob that exists in the repository and may be used by subsequent snapshots, even though the index will not be included in the snapshot being finalized. Otherwise, it is tracked in
```